src/option.c

```diff
@@ -4924,6 +4925,8 @@ check_num_option_bounds(
 	errmsg = e_invalid_argument;
 	p_mhi = 10000;
     }
+    check_msg_hist();
```

Comment:
This shouldn't be added here. Add a new `did_set_msghistory()` function and set it in `optiondefs.h`

Reply:
Thanks.  I have added it.